src/components/Forms/Health/RedisHealthFormEditor.tsx

```diff
@@ -0,0 +1,85 @@
+import FormikTextInput from "../Formik/FormikTextInput";
+import FormikTemplateFields from "../Formik/FormikTemplateFields";
+import FormikAuthFieldsGroup from "../Formik/FormikAuthFieldsGroup";
+import FormikCheckboxFieldsGroup from "../Formik/FormikCheckboxFieldsGroup";
+import FormikIconPicker from "../Formik/FormikIconPicker";
+import { getIn, useFormikContext } from "formik";
+import { useEffect } from "react";
+import FormikScheduleField from "../Formik/FormikScheduleField";
+
+type RedisHealthFormEditorProps = {
+  fieldName: string;
+  specsMapField: string;
+};
+
+export function RedisHealthFormEditor({
+  fieldName: name,
+  specsMapField
+}: RedisHealthFormEditorProps) {
+  const { values, setFieldValue } = useFormikContext();
+
+  const fieldName = `${name}.${specsMapField}`;
+
+  const nameValue = getIn(values, `${fieldName}.name`);
+
+  // when name changes, we want to update the name of the top level field
+  useEffect(() => {
+    setFieldValue("name", nameValue);
+  }, [nameValue, setFieldValue]);
+
+  return (
+    <>
+      <div className="flex flex-row md:space-x-2">
+        <FormikTextInput
+          className="flex flex-col w-1/2"
+          name={`${fieldName}.name`}
+          label="Name"
+          required
+        />
+        <FormikIconPicker
+          className="flex flex-col w-1/2"
+          name={`${fieldName}.icon`}
+          label="Icon"
+        />
+      </div>
+
+      <FormikScheduleField name={`${name}.schedule`} />
+
+      <FormikTextInput name={`${fieldName}.description`} label="Description" />
+
+      <FormikTextInput
```

Comment:
May be make the hint more descriptive or give an example.

![image](https://github.com/flanksource/flanksource-ui/assets/12270550/d0210878-d49a-4d25-af8b-c862876374e2)

Reply:
Can you suggest a better description ?